Sources/TensorFlow/Core/ShapedArray.swift

```diff
@@ -86,15 +86,10 @@ extension TensorBuffer {
         count: Int,
         withInitializer body: (UnsafeMutableBufferPointer<Scalar>) -> Void
     ) -> TensorBuffer<Scalar> {
-        /// Since `Scalar` may be any generic type, it is not possible to construct
-        /// an instance of `Scalar` directly for use with the
-        /// `Array(repeating:count:)` initializer. The workaround here is to
-        /// allocate a dummy `Scalar` pointer of size 1 and to use the pointee value
-        /// as the `repeatedValue` of the initializer.
-        let dummyPointer = UnsafeMutablePointer<Scalar>.allocate(capacity: 1)
-        var array = Array(repeating: dummyPointer.move(), count: count)
-        array.withUnsafeMutableBufferPointer { body($0) }
-        dummyPointer.deallocate()
+        let array = Array<Scalar>(unsafeUninitializedCapacity: 1) { buffer, initializedCount in
```

Comment:
The capacity here is incorrect. The buffer under `dummyPointer` was allocated with capacity `1` because we were to load its value (an uninitialized `Scalar`) and repeat it `count` times to initialize the array. Now that we are to allocate the array directly, we should allocate the array with a capacity equal to the number of elements we want to add to the array, aka. `count`.

```suggestion
        let array = Array<Scalar>(unsafeUninitializedCapacity: count) { buffer, initializedCount in
```

Reply:
Ah sorry about that @rxwei. I was confused about that and should've asked in the issue. Changing now.